cloudferrylib/os/compute/usage_quota.py

```diff
@@ -0,0 +1,187 @@
+        sql_render = sql.format(sec_gr_id=security_group_id,
+                                new_user_id=new_user_id,
+                                sec_groups=cls.NOVA_SEC_GRS)
+        LOG.info("Change owner for security groups %s", security_group_id)
+        LOG.debug(sql_render)
+        db.execute(sql_render)
+
+    @classmethod
+    def dec_quota_usage(cls, db, user_id, tenant_id, quota):
+        sql = "UPDATE {quota_usages} SET " \
+              "{quota_usages}.in_use = {quota_usages}.in_use - " \
+              "'{in_use}' WHERE ({quota_usages}.user_id='{user_id}')" \
+              " and ({quota_usages}.resource='{resource}') and " \
+              "({quota_usages}.project_id='{tenant_id}')"
+        LOG.info("Decrement usages quota user '%s'", user_id)
+        for res, value in quota.iteritems():
+            sql_render = sql.format(quota_usages=cls.NOVA_QUOTA_USAGES,
+                                    in_use=value, user_id=user_id,
+                                    tenant_id=tenant_id, resource=res)
+            LOG.debug(sql_render)
+            db.execute(sql_render)
+
+    @classmethod
+    def inc_quota_usage(cls, db, user_id, tenant_id, quota):
+        sql = "UPDATE {quota_usages} SET " \
+              "{quota_usages}.in_use = {quota_usages}.in_use + " \
+              "'{in_use}' WHERE ({quota_usages}.user_id='{user_id}')" \
+              " and ({quota_usages}.resource='{resource}') and " \
+              "({quota_usages}.project_id='{tenant_id}')"
+        LOG.info("Increment usages quota user '%s'", user_id)
+        for res, value in quota.iteritems():
+            if not cls.check_resource(db, tenant_id, user_id, res):
+                cls.insert_resource_usage(db, tenant_id, user_id, res)
+            sql_render = sql.format(quota_usages=cls.NOVA_QUOTA_USAGES,
+                                    in_use=value, user_id=user_id,
+                                    tenant_id=tenant_id, resource=res)
+            LOG.debug(sql_render)
+            db.execute(sql_render)
+
+    @classmethod
+    def check_resource(cls, db, tenant_id, user_id, resource):
+        sql = "SELECT COUNT(*) FROM {quota_usages} WHERE " \
+              "({quota_usages}.project_id='{tenant_id}') and " \
+              "({quota_usages}.user_id='{user_id}') and " \
+              "({quota_usages}.resource='{resource}')"\
+            .format(quota_usages=cls.NOVA_QUOTA_USAGES,
+                    tenant_id=tenant_id,
+                    user_id=user_id,
+                    resource=resource)
+        LOG.info("Check resource .... ")
+        LOG.debug(sql)
+        res = db.execute(sql)
+        return res.scalar()
+
+    @classmethod
+    def insert_resource_usage(cls, db, tenant_id, user_id, resource):
+        sql = "INSERT INTO {quota_usages}({columns}) VALUES ({values})"
+        columns = "created_at,updated_at,project_id," \
+                  "resource,in_use,deleted,user_id"
```

Comment:
There also should be 'reserved' column with 0 value, because it doesn't have default value.